app/flatpak-builtins-create-usb.c

```diff
@@ -605,21 +610,22 @@ flatpak_builtin_create_usb (int argc, char **argv, GCancellable *cancellable, GE
         const char *commit;
         CommitAndSubpaths *c_s;
 
-        deploy_data = flatpak_dir_get_deploy_data (dir, first_ref, cancellable, error);
+        deploy_data = flatpak_dir_get_deploy_data (dir, installed_ref, cancellable, error);
         if (deploy_data == NULL)
           return FALSE;
         commit = flatpak_deploy_data_get_commit (deploy_data);
         c_s = commit_and_subpaths_new (commit, NULL);
 
         g_hash_table_insert (all_collection_ids, g_strdup (ref_collection_id), g_strdup (remote));
-        collection_ref = ostree_collection_ref_new (ref_collection_id, first_ref);
+        collection_ref = ostree_collection_ref_new (ref_collection_id, installed_ref);
         g_hash_table_insert (all_refs, g_steal_pointer (&collection_ref), c_s);
       }
 
       /* Add dependencies and related refs */
-      if (!add_runtime (all_refs, all_collection_ids, first_ref, dir, cancellable, error))
+      if (!(installed_ref_kind & FLATPAK_KINDS_RUNTIME) &&
+          !add_runtime (all_refs, all_collection_ids, installed_ref, dir, cancellable, error))
         return FALSE;
-      if (!add_related (all_refs, all_collection_ids, first_ref, dir, cancellable, error))
+      if (!add_related (all_refs, all_collection_ids, installed_ref, dir, cancellable, error))
```

Comment:
Oh, I initially read this as `add_runtime()` so this looked like duplicated code but it isn't.